submissions/angels_unit_tests/angels_unit_tests.md

```diff
@@ -0,0 +1,55 @@
+
+This script saved so many time, allowing for a sporadic coffee break while it reloaded the mods after tweaking that one line that still didn't behave as intended. By using this script over time, I had another issue, Bob. Yes, Bobbingabout released an update of his mods, and it broke all the workd I just did fixing a bug... On the 3rd of May 2021, almost a year later, I created [another small python script](https://github.com/Arch666Angel/mods/pull/591/commits/3d0a7de0a68c3ec6566cbd45c3599b036ab775bb) to download all bob mods from the portal, rather than doing this ingame one by one.
+
+## The ball sped up rolling downhill
+
+At the end of 2020, we decided to work on a big feature inside Angel's Industries: a decent implementation of the component and technology overhaul. This was the last feature Arch666Angel was working on before he pulled back from modding. On the 26th of May 2021 (half a year since the last release), we presented the new version of the overhauls to the public, after which the bug reports started rolling in quickly.
+
+We realised that with the overhauls we created many more configuration, with even more settings. At this point it becomes quite though to maintain and test this completely by hand. I started thinking about creating some [unit tests](https://en.wikipedia.org/wiki/Unit_testing) to take this burden of our shoulders. After some discussions and ideas from the great modding community I finally decided on the 30th of June 2021 to commit myself at creating the unit test infrastructure, groupted in a separate mod: [Angel's Dev - Unit Tests](https://github.com/Arch666Angel/mods/pull/634/commits/d59328e81235e26beb88ba66659cb78315827a00). A small note, You will not find this mod on the modportal.
+
+I already had a way of building the Angel's mods from the github folder to the factorio mod folder and a similar script to download bob mods from the mod portal. In order to test different configurations, I had to write two additional scripts. [The first script](https://github.com/Arch666Angel/mods/pull/634/commits/8977e17bba55d06fb3b60ff4c593e2307f110395) reads and rewrites the `mod-settings.dat` file with the mod settings. Since this was a part that I didn't encounter in creating mods before, I was very surprised at finding [thorough documentation on the Factorio Wiki](https://wiki.factorio.com/Mod_settings_file_format) about this. [The second script](https://github.com/Arch666Angel/mods/pull/634/commits/abb64c2fe8007e060f112716ebe33864510b6451) configures the `mod-list.json` file, which contains the information about which mods have to be enabled.
```

Comment:
Maybe you can qive quick one-sentence explanations of what's stored in`mod-settings.dat` and `mod-list.json` and why you would need to modify them.

Reply:
Fixed in https://github.com/AlternativeFFFF/Alt-F4/pull/551/commits/0d2bc1d8be83f76d557f22d7d39ba53b96e0aa9f